server/src/projects/projects.controller.ts

```diff
@@ -6,14 +6,21 @@ import {
   Patch,
   Param,
   Delete,
+  UseGuards,
 } from '@nestjs/common';
 import { ProjectsService } from './projects.service';
 import { CreateProjectDto, UpdateProjectDto } from './dto';
+import { Roles } from 'src/auth/decrator';
+import { Role } from 'src/auth/enums/role.enum';
+import { JwtAuthGuard } from 'src/auth/strategy';
+import { RolesGuard } from 'src/auth/Guards/roles.guard';
 
 @Controller('projects')
 export class ProjectsController {
   constructor(private readonly projectsService: ProjectsService) {}
 
+  @Roles(Role.Admin)
+  @UseGuards(JwtAuthGuard, RolesGuard)
```

Comment:
I think first should check if there JWT token and Roles then chack if the role === admin 

- so the code should be in this order

@UseGuards(JwtAuthGuard, RolesGuard)
@Roles(Role.Admin)

Reply:
Actually what Basil did here is totally fine, because the "@Roles(Role. Admin)" decorator doesn't validate the request, it just adds some metadata (role in this case) to the controller. And this metadata is required for RolesGuard class.
So RolesGuard will extract the role from the request and compare it to the role in the metadata of the controller and if they are matched the controller will handle the request.
In summary:-  
@roles(Role.Admin) >> will define a role for the controller.
@RolesGuard >> will validate the role in the request to see if it matches the defined role.